plugins/modules/pacman_key.py

```diff
@@ -150,43 +193,67 @@ def __init__(self, module):
         keyid = self.sanitise_keyid(keyid)
-        key_present = self.key_in_keyring(keyring, keyid)
+        key_validity = self.key_validity(keyring, keyid)
+        key_present = len(key_validity) > 0
+        key_valid = any(key_validity)
 
         # check mode
         if module.check_mode:
-            if state == "present":
+            if state in ['present', 'trusted']:
                 changed = (key_present and force_update) or not key_present
+                if not changed and state == 'trusted':
+                    changed = not (key_valid and self.key_is_trusted(keyring, keyid))
                 module.exit_json(changed=changed)
-            elif state == "absent":
-                if key_present:
-                    module.exit_json(changed=True)
-                module.exit_json(changed=False)
-
-        if state == "present":
-            if key_present and not force_update:
-                module.exit_json(changed=False)
-
+            if state == 'absent':
+                module.exit_json(changed=key_present)
+
+        if state in ['present', 'trusted']:
+            trusted = key_valid and self.key_is_trusted(keyring, keyid)
+            if not force_update:
+                if (state == 'present' and key_present) or (state == 'trusted' and trusted):
+                    module.exit_json(changed=False)
+            changed = False
             if data:
                 file = self.save_key(data)
                 self.add_key(keyring, file, keyid, verify)
-                module.exit_json(changed=True)
+                changed = True
             elif file:
                 self.add_key(keyring, file, keyid, verify)
-                module.exit_json(changed=True)
+                changed = True
             elif url:
                 data = self.fetch_key(url)
                 file = self.save_key(data)
                 self.add_key(keyring, file, keyid, verify)
-                module.exit_json(changed=True)
+                changed = True
             elif keyserver:
                 self.recv_key(keyring, keyid, keyserver)
-                module.exit_json(changed=True)
-        elif state == "absent":
+                changed = True
+            if changed or (state == 'trusted' and not trusted):
+                self.lsign_key(keyring=keyring, keyid=keyid)
+                changed = True
+            module.exit_json(changed=changed)
+        elif state == 'absent':
             if key_present:
                 self.remove_key(keyring, keyid)
                 module.exit_json(changed=True)
             module.exit_json(changed=False)
 
+    def gpg(self, args, keyring=None, **kwargs):
+        cmd = [self.gpg_binary]
+        if keyring:
+            cmd.append('--homedir={keyring}'.format(keyring=keyring))
+        cmd.extend(['--no-permission-warning', '--with-colons', '--quiet', '--batch', '--no-tty'])
+        return self.module.run_command(cmd + args, **kwargs)
```

Comment:
You might want to consider using `CmdRunner`. See https://docs.ansible.com/ansible/latest/collections/community/general/docsite/guide_cmdrunner.html for more details.

Reply:
I started adding that, but at least for `gpg` I think it decreases readability due to the  `cmd_runner_fmt.as_fixed` args bunch. It also didn't seem likely for it to catch/prevent bugs, so I left it out, but let me know if I got that wrong.